src/Fixer/ControlStructure/NoUnusedCapturingCatchFixer.php

```diff
@@ -0,0 +1,155 @@
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ControlStructure;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Tokenizer\Analyzer\FunctionsAnalyzer;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixer\Tokenizer\TokensAnalyzer;
+
+/**
+ * @author SpacePossum
+ */
+final class NoUnusedCapturingCatchFixer extends AbstractFixer
+{
+    /**
+     * @var FunctionsAnalyzer
+     */
+    private $functionAnalyzer;
+
+    /**
+     * @var TokensAnalyzer
+     */
+    private $tokensAnalyzer;
+
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Remove not used captured exception variables. Requires PHP >= 8.0.',
+            [
+                new VersionSpecificCodeSample(
+                    "<?php\ntry {    foo();\n} catch (\\Exception \$e) {\n    // ignore exception\n}\n",
+                    new VersionSpecification(80000)
+                ),
+            ]
+        );
+    }
+
+    public function isCandidate(Tokens $tokens)
+    {
+        return \PHP_VERSION_ID >= 80000 && $tokens->isAllTokenKindsFound([T_CATCH, T_VARIABLE]);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $this->functionAnalyzer = new FunctionsAnalyzer();
+        $this->tokensAnalyzer = new TokensAnalyzer($tokens);
+
+        for ($index = \count($tokens) - 1; $index > 1; --$index) {
+            if ($tokens[$index]->isGivenKind(T_CATCH)) {
+                $this->fixCatch($tokens, $index);
+            }
+        }
+    }
+
+    /**
+     * @param int $index
+     */
+    private function fixCatch(Tokens $tokens, $index)
+    {
+        $index = $startIndex = $tokens->getNextTokenOfKind($index, ['(']);
+
+        do {
+            $index = $tokens->getNextMeaningfulToken($index);
+        } while ($tokens[$index]->isGivenKind([T_NS_SEPARATOR, T_STRING, CT::T_TYPE_ALTERNATION]));
+
+        if (!$tokens[$index]->isGivenKind(T_VARIABLE)) {
+            return;
+        }
+
+        if ($this->tokensAnalyzer->isSuperGlobal($index)) {
+            return;
+        }
+
+        $startIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $startIndex);
+        $startIndex = $tokens->getNextTokenOfKind($startIndex, ['{']);
+        $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $startIndex);
+
+        if (!$this->isCapturePossiblyUsed($tokens, $index, $startIndex, $endIndex)) {
+            $this->clearCapture($tokens, $index);
+        }
+    }
+
+    /**
+     * @param int $index
+     * @param int $startIndex
+     * @param int $endIndex
+     *
+     * @return bool
+     */
+    private function isCapturePossiblyUsed(Tokens $tokens, $index, $startIndex, $endIndex)
+    {
+        static $riskyKinds = [
```

Comment:
Duplicated logic with the unused closure use fixer?

Reply:
:+1:  good point, extracted, continuing the work now